python/dgl/distributed/graph_partition_book.py

```diff
@@ -21,15 +20,23 @@
     _to_shared_mem,
 )
 
-
-def _str_to_tuple(s):
-    try:
-        ret = literal_eval(s)
-    except ValueError:
-        ret = s
+CANONICAL_ETYPE_DELIMITER = ":"
+
+def _etype_tuple_to_str(c_etype):
+    '''Convert canonical etype from tuple to string.
+    '''
+    assert isinstance(c_etype, tuple) and len(c_etype) == 3, \
+        "Passed-in canonical etype should be in format of (str, str, str)."
```

Comment:
`"Passed-in ... . But got {c_etype}."`

Printing out the value will be more useful.